package.json

```diff
@@ -76,7 +76,8 @@
     "dist"
   ],
   "scripts": {
-    "build": "tsup",
+    "build": "pnpm build:types && tsup",
+    "build:types": "pnpm vue-tsc --declaration --emitDeclarationOnly -p tsconfig.build-types.json && rimraf src/core/components.d.ts",
```

Comment:
the declaration omitted from `Story` seems strange. Would you take another look?

Does it have to do with the external types needing to be added?

```ts
// Due to a TypeScript bug (https://github.com/microsoft/TypeScript/issues/47663), we need to export types from dependencies
export type { StoryObj } from '@storybook/vue3'
export type { ArgTypes, Args, Parameters, StrictArgs } from '@storybook/types'
export type {
  Simplify,
  SetOptional,
  Constructor,
  RemoveIndexSignature,
} from 'type-fest'
```

I suppose this is due to the ` This is likely not portable. A type annotation is necessary.`

This is because `story.vue` is importing

```ts
import type { Meta, StoryObj } from '@storybook/vue3'
```

(I am stating the obvious here as I keep track of my thoughts).

All of this is just needed because of `play` option on a Story. Does it already work?

Do we have documentation of that API?

Otherwise, I would consider removing it from the current scope, so that we can also simplify this export until we've mapped out the missing pieces.